src/VisualizationObject.ts

```diff
@@ -131,39 +148,57 @@ export namespace VisualizationObject {
     }
 
     export function isMeasure(bucketItem: IMeasure | IVisualizationAttribute): bucketItem is IMeasure {
-        return (bucketItem as IMeasure).measure !== undefined;
+        return !isEmpty(bucketItem) && (bucketItem as IMeasure).measure !== undefined;
     }
 
     export function isVisualizationAttribute(
         bucketItem: IMeasure | IVisualizationAttribute
     ): bucketItem is IVisualizationAttribute {
-            return (bucketItem as IVisualizationAttribute).visualizationAttribute !== undefined;
+        return !isEmpty(bucketItem) && (bucketItem as IVisualizationAttribute).visualizationAttribute !== undefined;
     }
 
     export function isMeasureDefinition(
-        definition: IMeasureDefinition | IPoPMeasureDefinition
+        definition: IMeasureDefinitionType
     ): definition is IMeasureDefinition {
-        return (definition as IMeasureDefinition).measureDefinition !== undefined;
+        return !isEmpty(definition) && (definition as IMeasureDefinition).measureDefinition !== undefined;
+    }
+
+    export function isPopMeasureDefinition(
+        definition: IMeasureDefinitionType
+    ): definition is IPoPMeasureDefinition {
+        return !isEmpty(definition) && (definition as IPoPMeasureDefinition).popMeasureDefinition !== undefined;
+    }
+
+    export function isPreviousPeriodMeasureDefinition(
+        definition: IMeasureDefinitionType
+    ): definition is IPreviousPeriodMeasureDefinition {
+        return !isEmpty(definition)
+            && (definition as IPreviousPeriodMeasureDefinition).previousPeriodMeasure !== undefined;
     }
 
     export function isAttributeFilter(filter: VisualizationObjectFilter): filter is VisualizationObjectAttributeFilter {
-        return (filter as IVisualizationObjectPositiveAttributeFilter).positiveAttributeFilter !== undefined ||
-            (filter as IVisualizationObjectNegativeAttributeFilter).negativeAttributeFilter !== undefined;
+        return !isEmpty(filter) && (
+            (filter as IVisualizationObjectPositiveAttributeFilter).positiveAttributeFilter !== undefined
+            || (filter as IVisualizationObjectNegativeAttributeFilter).negativeAttributeFilter !== undefined
+        );
     }
 
     export function isPositiveAttributeFilter(
         filter: VisualizationObjectAttributeFilter
-    ): filter is IVisualizationObjectPositiveAttributeFilter {
-        return (filter as IVisualizationObjectPositiveAttributeFilter).positiveAttributeFilter !== undefined;
+    ):
+        filter is IVisualizationObjectPositiveAttributeFilter {
```

Comment:
This should probably be on the previous line.

Reply:
done